### PR TITLE
Fix test login (maybe)

### PR DESCRIPTION
**Pull request by solderq35**
## Issue

Original Error (running `yarn test`)
image

In addition to the above error:
- Before, if you logged in as a regular user (running `yarn start`), didn't log out afterwards, then ran the app again as `yarn test`, then it tries to log you in as a regular user again instead of test user (due to persistent login code conflicting with the test user code)

## Changes (still applicable to latest commit of this PR)
Tweaked auth logic to prevent persisting the regular user login if running in test mode (`yarn test`).

Other changes to drill input is just what happened from running `yarn pretty`

## Troubleshooting
- Deleting the contents of `node_modules` folder (and regenerate them with `yarn`) may also help with preventing the error, not sure (regardless, the changed code should still be useful so that the test user works without having to log out the real user first).
- Also try running app normally as `yarn start` and logging out of regular user
- Clear yarn cache with `yarn start -c` or `yarn test -c` (suggested by Frank)

## Other potential features
- In the future it could be worth just adding a test login button rather than relying on environmental variables if it's easier to debug / more intuituive

### CODE REVIEW

**Review by Gehrkej — COMMENTED** [review]
Going through the troubleshooting options, I wasn't able to get `yarn test` to work even when deleting and reinstalling `node_modules` as well as appending a `-c `when running it. The solution I found that worked was just using `yarn start` and creating an account. Although with this, I found there was already an account made using my ONID. To reset my password, I navigated to firebase project, went to authentication, found my email and utilized the reset password email functionality. 

I believe Jeff's changes look good for preventing persistent login but was unable to fully test as I kept running into the initial issue.

**Review by FrankreedX — CHANGES_REQUESTED** [review]

File: context/Auth.js [hunk 1]
```diff
@@ -45,15 +45,20 @@ export const AuthProvider = ({ children }) => {
   useEffect(() => {
     //if this code is not in here, it'll run for infinite times
     onAuthStateChanged(auth, (currentUserId) => {
+      // keep this log outside if statement to confirm e.g. logout of test user
       console.log("user changed");
-      if (currentUserId) {
-        setCurrentUserId(currentUserId.uid ?? "Error (uid)");
+
+      // prevent "persistent login" of regular user login from conflicting with test user
+      if (!process.env.EXPO_PUBLIC_TEST_UID) {
```
[the remark this answers is not in the document]
Comment (solderq35) [reply context/Auth.js]: I moved that if statement up, hopefully it's what you meant. Everything still seems to be working as expected

[the remark this answers is not in the document]
Comment (solderq35) [reply context/Auth.js]: Altho the implementation will probably change with the login button version

**Review by FrankreedX — APPROVED** [review]
it's fine

**Comment by solderq35** [comment]
image

Test Login Button Version (no async storage on the test user)

**Comment by solderq35** [comment]
# Final Version? (yarn test without error)
This is the final version for latest commit of this PR.
## Testing
Run the following in terminal
- `git fetch`
  - Optional, just to make sure you get all the new branches from remote if applicable
- `git checkout solderq35/persist-async-storage`
- `git pull`
  - Optional, only if not up to date with remote
- `yarn`
- `yarn test -c`
  - No need to actually run the app on your phone yet, but let the app finish compiling, then quit with Ctrl / Cmd C
- `git checkout layout`
- git pull`
  - Optional, only if not up to date with remote
- `yarn test`
  - This time, **scan the QR code with your phone**, which should confirm if the error triggers or not (`...Invalid call at line 2: process.env.EXPO_ROUTER_APP_ROOT... First argument of `require.context` should be a string denoting the directory to require.`)
  - The error should trigger here
- `git checkout solderq35/fix-test-login`
- git pull`
  - Optional, only if not up to date with remote
- `yarn test`
  - App should compile this time

Check my log file below in case I made a mistake in my instructions above
[reproduce-build-cache-bug.log](https://github.com/Golf-Drill-Challenge-App/Golf-App/files/14858795/reproduce-build-cache-bug.log)

## Explanation / Research (summary)
Git Branch differences
- `solderq35/persist-async-storage` changes some stuff in `app.json` to account for persistent caching
  - As explained by Expo dev Evan Bacon in [this thread](https://github.com/expo/expo/issues/23819#issuecomment-2019198302), changes to `app.json` may cause issues, which might resolved by clearing the bundler cache
    - However, this did not fully explain the bug at least in our use case, because Jake reported that sometimes clearing the cache `yarn test -c` did not fix the bug, and I also found that sometimes clearing the cache with `-c` was not enough
      - **NOTE: It seemed like another (relatively) fail-safe method was to add or remove some arbitrary package to package.json, then run `yarn`, then do `yarn test -c`**
        - This is why I intentionally only ran `yarn` on `solderq35/persist-async-storage` branch, and not on `layout`, nor `solderq35/fix-test-login` branch. Since `persist-async-storage` branch has package differences compared to `layout` and `fix-test-login` branches, running `yarn` on `persist-async-storage` along with `yarn test -c` "reset" the build cache onto the `persist-async-storage` branch, and caused the build cache to expect the app.json of `persist-async-storage` (theory)
        - It's also why I made sure that `babel-plugin-transform-inline-environment-variables` library was in both `persist-async-storage` and `fix-test-login` branches, so that the `fix-test-login` branch would at least compile without needing to run `yarn` to reinstall packages again
- The other half of the puzzle was the solution from this [other thread](https://github.com/expo/router/issues/41#issuecomment-1657674119) to make edits to `babel.config.js` to resolve issues with expo root / babel, and to add `babel-plugin-transform-inline-environment-variables` library
  - **For future reference, we seem to still be on Expo SDK 49 `"expo": "~49.0.15"` in package.json (which is why the first thread I linked from Evan Bacon is about SDK 49), so the babel / expo root / environmental variables stuff may need to be updated if we ever move to Expo SDK 50 or higher**
  - **Although it is generally still advised to clear the build cache with `-c`, it seemed the bug was more closely related to Babel and the "expo root" having some kind of conflict, after changes to app.json**

**Review by FrankreedX — APPROVED** [review]
sick. I have tested this on my machine and it works

**Review by hannacol — APPROVED** [review]

**Pull request merged**